### PR TITLE
Bug 1252235: add a shellquote filter

### PR DESCRIPTION
**Pull request by djmitche**
This would be used like this:

```
  --message={{#shellquote}}{{{comment}}}{/shellquote}}
```

Aside from deploying to staging, I don't have a good idea of how to test this..

### CODE REVIEW

File: src/jobs/taskcluster_graph.js [hunk 1]
```diff
@@ -86,6 +87,12 @@ export default class TaskclusterGraphJob extends Base {
     let level = projectConfig.level(this.config.try, repo.alias);
     let scopes = projectConfig.scopes(this.config.try, repo.alias)
 
+    // quote the contained text for passing to the shell; note that this includes
+    // the leading and trailing " or ' characters, so they must not be included
+    // in the template.
+    let shellquote = (text, render) {
```
Comment (gregarndt) [inline src/jobs/taskcluster_graph.js]: are you missing the fat arrow here?

**Comment by gregarndt** [comment]
you should be able to test this on staging.

**Comment by djmitche** [comment]
I don't always test my code, but when I do, I do it in production

Staging doesn't create task-graphs, so pushing to prod is probably the most effective way to test this.

**Comment by djmitche** [comment]
Well, that produced

```
Aug 31 16:20:25 mozilla-taskcluster app/worker.1: Wed, 31 Aug 2016 21:20:25 GMT superagent end GET https://hg.mozilla.org/try/raw-file/ebff045513b0c3362016ffe002473d18367e9792/.taskcluster.yml 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1: Error creating graph due to yaml syntax errors, { 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "owner": "ahalberstadt@mozilla.com", 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "revision": "ebff045513b0c3362016ffe002473d18367e9792", 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "project": "try", 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "level": 1, 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "revision_hash": "648989a3e665cf72bd9de781a47b76e63f30dad3", 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "comment": "try: -b o -p win32,linux64,macosx64 -u none -t none", 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "pushlog_id": "142051", 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "url": "https://hg.mozilla.org/try/", 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "importScopes": true, 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "pushdate": 1472678380, 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:   "source": "https://hg.mozilla.org/try/raw-file/ebff045513b0c3362016ffe002473d18367e9792/.taskcluster.yml" 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1: } 
Aug 31 16:20:25 mozilla-taskcluster app/worker.1:  
Aug 31 16:20:25 mozilla-taskcluster app/worker.1: [1] shellquote is not allowed 
Aug 31 16:20:25 mozilla-taskcluster app/pushlog.1: Wed, 31 Aug 2016 21:20:25 GMT superagent end GET https://hg.mozilla.org/projects/cypress/json-pushes/ 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1: Error fetching .taskcluster.yml at https://hg.mozilla.org/try/raw-file/ebff045513b0c3362016ffe002473d18367e9792/.taskcluster.yml for try. ReferenceError: job is not defined 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1:     at TaskclusterGraphJob.scheduleTaskGroup$ (/app/src/jobs/taskcluster_graph.js:178:41) 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1:     at tryCatch (/app/node_modules/regenerator-babel/runtime.js:53:40) 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1:     at GeneratorFunctionPrototype.invoke (/app/node_modules/regenerator-babel/runtime.js:209:22) 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1:     at tryCatch (/app/node_modules/regenerator-babel/runtime.js:53:40) 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1:     at Function.step (/app/node_modules/regenerator-babel/runtime.js:103:22) 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1:     at /app/node_modules/regenerator-babel/runtime.js:117:7 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1:     at Object.runtime.async (/app/node_modules/regenerator-babel/runtime.js:97:12) 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1:     at TaskclusterGraphJob.scheduleTaskGroup (/app/src/jobs/taskcluster_graph.js:205:21) 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1:     at TaskclusterGraphJob.work$ (/app/src/jobs/taskcluster_graph.js:138:25) 
Aug 31 16:20:27 mozilla-taskcluster app/worker.1:     at tryCatch (/app/node_modules/regenerator-babel/runtime.js:53:40) 
```

So there's a bug, and another bug in the error handling (job is not defined).

**Comment by djmitche** [comment]
I reverted and reset the pushids back, so the two task-graphs which failed to create would be re-tried, but that appears not to have worked, and both branches have now had additional pushes with successful decision tasks.

**Comment by gregarndt** [comment]
If you reset the push id I think you need to restart the app (in case you reset the id after reverting).  

Looks like it's the nasty stuff in here probably https://github.com/taskcluster/mozilla-taskcluster/blob/master/src/try/instantiate.js#L62-L74

**Pull request merged**